### PR TITLE
Add device handle member to bno055_t struct from use in read and write callback functions

### PR DESCRIPTION
**Pull request by bdesterBE**

### CODE REVIEW

**Review by cmorganBE — CHANGES_REQUESTED** [review]
I'd say why this commit is being done in the subject, like:

"Pass user data back to callback functions to enable supporting multiple bno005 devices"

And the commit body could be:

"Add device handle member to bno055_t struct from use in read and write callback function"

File: bno055.h [hunk 1]
```diff
@@ -521,6 +521,7 @@ struct bno055_t
     BNO055_WR_FUNC_PTR; /**< bus write function pointer */
     BNO055_RD_FUNC_PTR; /**<bus read function pointer */
     void (*delay_msec)(BNO055_MDELAY_DATA_TYPE); /**< delay function pointer */
+    void *dev_handle; /**< device function pointer */
```
Comment (cmorganBE) [inline bno055.h]: can we make this 'user_data'?

Comment (cmorganBE) [reply bno055.h]: and clarify that its used to hold context, like /**< User data pointer */

Comment (bdesterBE) [reply bno055.h]: changed to user_data

**Review by cmorganBE — APPROVED** [review]

**Review by gseqBE — APPROVED** [review]

**Pull request merged**